spec/models/course/assessment/answer/answer_ability_spec.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+RSpec.describe Course::Assessment::Answer do
+  let!(:instance) { Instance.default }
+  with_tenant(:instance) do
+    subject { Ability.new(user, course, course_user) }
+    let(:course) { create(:course) }
+    let(:course_user) { create(:course_student, course: course) }
+    let(:coursemate) { create(:course_student, course: course) }
+    let(:fs_assessment) do
+      create(:assessment, :published, :with_programming_file_submission_question, course: course)
+    end
+    let(:fs_submission) { create(:submission, :attempting, assessment: fs_assessment, creator: course_user.user) }
+    let(:programming_assessment) do
+      create(:assessment, :published_with_programming_question, course: course)
+    end
+    let(:programming_submission) { create(:submission, :attempting, assessment: programming_assessment, creator: user) }
+    let(:mcq_assessment) do
+      create(:assessment, :published_with_mcq_question, course: course)
+    end
+    let(:mcq_submission) { create(:submission, :attempting, assessment: mcq_assessment, creator: user) }
+    let(:coursemate_fs_submission) do
+      create(:submission, :attempting, assessment: fs_assessment, creator: coursemate.user)
+    end
+
+    context 'when the user is a Course Student' do
```

Comment:
You need to test the case when we cannot update the submission as well. I think it is left out here.

Reply:
Shouldn't this be under the `submission_ability` instead?

Reply:
We should also test whether students can destroy or create programming files based on the submission ability. If the tests are not written here, the removal of the `can_update_submission?` in [this line](https://github.com/Coursemology/coursemology2/pull/2436/files#diff-669b28d7cc8a04723ab0b0266a7f24c7R14) will not be tested, which can be tricky in the future during refactoring.